diff --git a/configure b/configure
index 3d5e3b6..abc210f 100755
--- a/configure
+++ b/configure
@@ -13369,7 +13369,6 @@ else
 esac
 
 
-$as_echo "#define malloc rpl_malloc" >>confdefs.h
 
 fi
 
@@ -13436,7 +13435,6 @@ else
 esac
 
 
-$as_echo "#define realloc rpl_realloc" >>confdefs.h
 
 fi
 
diff --git a/libGeoIP/GeoIP.h b/libGeoIP/GeoIP.h
index 2852595..c0986a3 100644
--- a/libGeoIP/GeoIP.h
+++ b/libGeoIP/GeoIP.h
@@ -154,18 +154,8 @@ typedef enum {
     GEOIP_CORPORATE_SPEED = 3,
 } GeoIPNetspeedValues;
 
-#if defined(_WIN32) && !defined(__MINGW32__)
-#ifdef GEOIP_EXPORTS
-#define GEOIP_API __declspec(dllexport)
-#define GEOIP_DATA __declspec(dllexport)
-#else
-#define GEOIP_DATA __declspec(dllimport)
-#define GEOIP_API
-#endif /* GEOIP_EXPORTS */
-#else
-#define GEOIP_API
 #define GEOIP_DATA
-#endif
+#define GEOIP_API
 
 extern char **GeoIPDBFileName;
 extern GEOIP_DATA const char *GeoIPDBDescription[NUM_DB_TYPES];
diff --git a/libGeoIP/pread.h b/libGeoIP/pread.h
index 4e48949..7b5505a 100644
--- a/libGeoIP/pread.h
+++ b/libGeoIP/pread.h
@@ -24,4 +24,4 @@ typedef int ssize_t;
 int pread(int fd, void *buf, unsigned int nbyte, long offset);
 #endif
 
-#define HAVE_PREAD
+//#define HAVE_PREAD
