diff --git a/doc/Makefile.in b/doc/Makefile.in
index da86692..6251b89 100644
--- a/doc/Makefile.in
+++ b/doc/Makefile.in
@@ -412,7 +412,7 @@ am__define_uniq_tagged_files = \
   done | $(am__uniquify_input)`
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = examples scripts credentials latex reference
+DIST_SUBDIRS = scripts credentials latex reference
 am__DIST_COMMON = $(gnutls_TEXINFOS) $(srcdir)/Makefile.in \
 	$(top_srcdir)/build-aux/depcomp \
 	$(top_srcdir)/build-aux/mdate-sh \
@@ -1553,7 +1553,7 @@ IMAGES = \
 	gnutls-logo.png gnutls-x509.png	\
 	pkcs11-vision.png
 
-SUBDIRS = examples scripts credentials latex $(am__append_1)
+SUBDIRS = scripts credentials latex $(am__append_1)
 info_TEXINFOS = gnutls.texi gnutls-guile.texi
 
 # Examples.
@@ -1569,13 +1569,6 @@ gnutls_TEXINFOS = gnutls.texi fdl-1.3.texi cha-bib.texi \
 	cha-errors.texi cha-support.texi cha-shared-key.texi \
 	cha-gtls-examples.texi cha-upgrade.texi cha-tokens.texi \
 	cha-crypto.texi cha-auth.texi stamp_functions \
-	examples/ex-client-anon.c examples/ex-session-info.c \
-	examples/ex-verify.c examples/ex-cert-select.c \
-	examples/ex-client-resume.c examples/ex-client-srp.c \
-	examples/ex-client-x509.c examples/ex-serv-x509.c \
-	examples/ex-serv-anon.c examples/ex-serv-srp.c \
-	examples/ex-alert.c examples/ex-x509-info.c examples/ex-crq.c \
-	examples/ex-pkcs12.c examples/ex-client-dtls.c \
 	gnutls-internals.dia gnutls-internals.eps gnutls-internals.png \
 	gnutls-layers.dia gnutls-layers.eps gnutls-layers.png \
 	gnutls-crypto-layers.dia gnutls-crypto-layers.eps \
