diff --git a/include/m_string.h b/include/m_string.h
index a332e5a..783effa 100644
--- a/include/m_string.h
+++ b/include/m_string.h
@@ -132,13 +132,34 @@ static inline ulonglong my_strtoull(const char *nptr, char **endptr, int base)
 #endif
 }
 
-static inline char *my_strtok_r(char *str, const char *delim, char **saveptr)
+static inline char *my_strtok_r(char *str, const char *delim, char **nextp)
 {
-#if defined _WIN32
-  return strtok_s(str, delim, saveptr);
-#else
-  return strtok_r(str, delim, saveptr);
-#endif
+    char *ret;
+
+    if (str == NULL)
+    {
+        str = *nextp;
+    }
+
+    str += strspn(str, delim);
+
+    if (*str == '\0')
+    {
+        return NULL;
+    }
+
+    ret = str;
+
+    str += strcspn(str, delim);
+
+    if (*str)
+    {
+        *str++ = '\0';
+    }
+
+    *nextp = str;
+
+    return ret;
 }
 
 /* native_ rather than my_ since my_strcasecmp already exists */
